File: yangxj96-bean/src/main/java/com/yangxj96/saas/bean/security/Token.java
package com.yangxj96.saas.bean.security;

import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import lombok.*;

import java.io.Serializable;
import java.util.List;

/**
 * 响应的token实体
 */
@Data
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
class Token implements Serializable {

    /**
     * 账户ID
     */
    @JsonSerialize(using = ToStringSerializer.class)
    private Long id;

    /**
     * 用户名
     */
    private String username;

    /**
     * 访问 token
     */
    private String token;

    /**
     * 权限列表
     */
    private List<String> authorities;

    /**
     * 角色列表
     */
    private List<String> roles;

}

File: yangxj96-bean/src/main/kotlin/com/yangxj96/saas/bean/security/Token.kt
package com.yangxj96.saas.bean.security

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import java.io.Serializable

/**
 * 响应的token实体
 */
class Token : Serializable {

    /**
     * 账户ID
     */
    @JsonSerialize(using = ToStringSerializer::class)
    var id: Long? = null

    /**
     * 用户名
     */
    var username: String? = null

    /**
     * 访问 token
     */
    var token: String? = null

    /**
     * 权限列表
     */
    var authorities = mutableListOf<String>()

    /**
     * 角色列表
     */
    var roles = mutableListOf<String>()

    override fun toString(): String {
        return "Token(username=$username, token=$token, authorities=$authorities, roles=$roles)"
    }
}
